avm/res/event-hub/namespace/network-rule-set/main.bicep

```diff
@@ -0,0 +1,61 @@
+metadata name = 'Event Hub Namespace Network Rule Sets'
+metadata description = 'This module deploys an Event Hub Namespace Network Rule Set.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Conditional. The name of the parent event hub namespace. Required if the template is used in a standalone deployment.')
+param namespaceName string
+
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+@description('Optional. This determines if traffic is allowed over public network. Default is "Enabled". If set to "Disabled", traffic to this namespace will be restricted over Private Endpoints only and network rules will not be applied.')
+param publicNetworkAccess string = 'Enabled'
+
+@allowed([
+  'Allow'
+  'Deny'
+])
+@description('Optional. Default Action for Network Rule Set. Default is "Allow". It will not be set if publicNetworkAccess is "Disabled". Otherwise, it will be set to "Deny" if ipRules or virtualNetworkRules are being used.')
+param defaultAction string = 'Allow'
+
+@description('Optional. Value that indicates whether Trusted Service Access is enabled or not.')
+param trustedServiceAccessEnabled bool = true
+
+@description('Optional. An array of subnet resource ID objects that this Event Hub Namespace is exposed to via Service Endpoints. You can enable the `ignoreMissingVnetServiceEndpoint` if you wish to add this virtual network to Event Hub Namespace but do not have an existing service endpoint. It will not be set if publicNetworkAccess is "Disabled". Otherwise, when used, defaultAction will be set to "Deny".')
+param virtualNetworkRules array = []
+
+@description('Optional. An array of objects for the public IP ranges you want to allow via the Event Hub Namespace firewall. Supports IPv4 address or CIDR. It will not be set if publicNetworkAccess is "Disabled". Otherwise, when used, defaultAction will be set to "Deny".')
+param ipRules array = []
+
+var networkRules = [for (virtualNetworkRule, index) in virtualNetworkRules: {
+  ignoreMissingVnetServiceEndpoint: contains(virtualNetworkRule, 'ignoreMissingVnetServiceEndpoint') ? virtualNetworkRule.ignoreMissingVnetServiceEndpoint : null
+  subnet: contains(virtualNetworkRule, 'subnetResourceId') ? {
+    id: virtualNetworkRule.subnetResourceId
+  } : null
+}]
+
+resource namespace 'Microsoft.EventHub/namespaces@2022-10-01-preview' existing = {
+  name: namespaceName
+}
+
+resource networkRuleSet 'Microsoft.EventHub/namespaces/networkRuleSets@2022-10-01-preview' = {
+  name: 'default'
```

Comment:
Should be a parameter with default value `default` as is implemented e.g. for `fileServices`
https://github.com/Azure/bicep-registry-modules/blob/efdaf7b267af89252ff77105eac6b769ca29bb05/avm/res/storage/storage-account/file-service/main.bicep#L9-L10

Reply:
updated